lib/rubocop/ast/node/arg_node.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module AST
+    # A node extension for `arg`, `optarg`, `restarg`, `kwarg`, `kwoptarg`,
+    # `kwrestarg`, `blockarg`, `shadowarg` and `forward_arg` nodes.
+    # This will be used in place of a plain node when the builder constructs
+    # the AST, making its methods available to all `arg` nodes within RuboCop.
+    class ArgNode < Node
+      # Returns the name of an argument.
+      #
+      # @return [Symbol, nil] the name of the argument
+      def name
+        node_parts[0]
+      end
+
+      # Returns the default value of the argument, if any.
+      #
+      # @return [Node, nil] the default value of the argument
+      def default_value
+        return unless optional?
+
+        node_parts[1]
+      end
+
+      # Checks whether the argument is optional (it has a default value)
+      #
+      # @return [Boolean] whether the argument is optional
+      def optional?
+        optarg_type? || kwoptarg_type?
+      end
+
+      # Checks whether the argument is a block argument
+      #
+      # @return [Boolean] whether the argument is a block argument
+      def block?
```

Comment:
I'm not sure that these aliases add much in terms of clarity. I realize you put lots of work in the tests, but I'm not convinced we need them and I fear they just add to the cognitive load (a reader might wonder... is that different from `blockarg_type?`?)

Reply:
Eh I'm not worried about removing things 😄 I was trying to give this a fluent interface, but the ones that are just essentially `*_type` aliases are probably not useful. I'll remove `block?`, `splatted?`, `double_splatted?` (which I didn't like as a name, hence the half thought out alias 😅) `shadow?` and `forwarding?`.